packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/scrollable_a11y.ts

```diff
@@ -0,0 +1,142 @@
+    this.rowsViewFocusOutHandlerContext = this.rowsViewFocusOutHandlerContext || this.rowsViewFocusOutHandler.bind(this);
+  }
+
+  protected subscribeToRowsViewFocusEvent(): void {
+    super.subscribeToRowsViewFocusEvent();
+
+    const $rowsView = this._rowsView?.element();
+
+    eventsEngine.on($rowsView, 'focusout', this.rowsViewFocusOutHandlerContext);
+  }
+
+  protected unsubscribeFromRowsViewFocusEvent(): void {
+    super.unsubscribeFromRowsViewFocusEvent();
+
+    const $rowsView = this._rowsView?.element();
+
+    eventsEngine.off($rowsView, 'focusout', this.rowsViewFocusOutHandlerContext);
+  }
+
+  protected rowsViewFocusHandler(event: any): void {
+    const $target = $(event.target);
+
+    this.translateFocusIfNeed(event, $target);
+
+    super.rowsViewFocusHandler(event);
+  }
+
+  private rowsViewFocusOutHandler(): void {
+    this.makeScrollableFocusableIfNeed();
+  }
+
+  private translateFocusIfNeed(event: any, $target: dxElementWrapper): void {
+    const needTranslateFocus = this.isScrollableNeedFocusable();
+    const isFirstCellFixed = this._isFixedColumn(0);
+
+    if (!needTranslateFocus || !isFirstCellFixed) {
+      return;
+    }
+
+    const $firstCell = this._rowsView.getCell({ rowIndex: 0, columnIndex: 0 });
+    const firstCellHasTabIndex = !!$firstCell.attr('tabindex');
+
+    // @ts-expect-error dxElementWrapper doesn't have overload for 'is' method
+    const notFixedCellIsTarget = $target.is(this._$firstNotFixedCell);
+
+    if (firstCellHasTabIndex && notFixedCellIsTarget) {
+      event.preventDefault();
+
+      this._focus($firstCell);
+    }
+  }
+
+  protected renderCompleted(e: any): void {
+    this._$firstNotFixedCell = this.getFirstNotFixedCell();
+    this.makeScrollableFocusableIfNeed();
+
+    super.renderCompleted(e);
+  }
+
+  protected _tabKeyHandler(eventArgs: any, isEditing: any): void {
+    const isCellPositionDefined = isDefined(this._focusedCellPosition)
+      && !isEmptyObject(this._focusedCellPosition);
+    const isOriginalHandlerRequired = !isCellPositionDefined
+      || (!eventArgs.shift && this._isLastValidCell(this._focusedCellPosition))
+      || (eventArgs.shift && this._isFirstValidCell(this._focusedCellPosition));
+
+    if (isOriginalHandlerRequired) {
+      if (this.isScrollableNeedFocusable()) {
+        this._$firstNotFixedCell?.removeAttr('tabIndex');
+      }
+    }
+
+    super._tabKeyHandler(eventArgs, isEditing);
+  }
+
+  private getFirstNotFixedCell(): dxElementWrapper | undefined {
+    const columns = this._columnsController.getVisibleColumns();
+
+    let columnIndex = -1;
+
+    for (let index = 0; index < columns.length; index += 1) {
+      const notFixedColumn = !columns[index].fixed;
+
+      if (notFixedColumn) {
+        columnIndex = index;
+        break;
+      }
+    }
+
+    if (columnIndex === -1) {
+      return undefined;
+    }
+
+    return this._rowsView._getCellElement(0, columnIndex) as dxElementWrapper | undefined;
```

Comment:
Let's make this function shorter:

```suggestion
    const columns = this._columnsController.getVisibleColumns();
    const columnIndex = columns.findIndex(({ fixed }) => !fixed);

    return columnIndex === -1
      ? undefined
      : this._rowsView._getCellElement(0, columnIndex) as dxElementWrapper | undefined;
```